AdyenCard/Components/Card/CardViewController.swift

```diff
@@ -301,7 +302,13 @@ extension FormValueItem where ValueType == String {
 
 extension CardViewController: CardViewControllerProtocol {
     func update(storePaymentMethodFieldVisibility isVisible: Bool) {
-        items.storeDetailsItem.value = false
+        if !isVisible {
+            items.storeDetailsItem.value = false
+        }
         items.storeDetailsItem.isVisible = isVisible
     }
+
+    func update(storePaymentMethodFieldValue isOn: Bool) {
+        items.storeDetailsItem.value = isOn
```

Comment:
With this we can always be sure that shopper give explicit consent:

```suggestion
        if !isVisible {
             items.storeDetailsItem.value = isOn
        }
```

Reply:
are you referring to this: [a536d8c](https://github.com/Adyen/adyen-ios/pull/875/commits/a536d8cfb27152031e6c4a52287abf2b63795dfd)

Reply:
Or is the idea here to also pass the component visibility?

Reply:
Sorry, I have updated `suggested change`
The idea here is to prevent storing shopper's payment without shopper's consent (visual awareness).

Reply:
done here [aedc3c8](https://github.com/Adyen/adyen-ios/pull/875/commits/aedc3c84eb778539962dcdc06465348e8294da20)